### PR TITLE
DOC: fix definition of optional `disp` parameter

### PR DESCRIPTION
**Pull request by sciunto**
I discovered a mistake in the documentation of _minimize_lbfgsb which is visible [here](https://docs.scipy.org/doc/scipy/reference/optimize.minimize-lbfgsb.html#optimize-minimize-lbfgsb).

### CODE REVIEW

File: scipy/optimize/lbfgsb.py [hunk 1]
```diff
@@ -218,8 +218,9 @@ def _minimize_lbfgsb(fun, x0, args=(), jac=None, bounds=None,
 
     Options
     -------
-    disp : bool
-       Set to True to print convergence messages.
+    disp : int
```
Comment (andyfaff) [inline scipy/optimize/lbfgsb.py]: None or int

File: scipy/optimize/lbfgsb.py [hunk 2]
```diff
@@ -218,8 +218,9 @@ def _minimize_lbfgsb(fun, x0, args=(), jac=None, bounds=None,
...
+        If zero, then no output.  If a positive number, then this over-rides
```
Comment (andyfaff) [inline scipy/optimize/lbfgsb.py]: If `disp is None` (the default), then the supplied version of `iprint` is used. If `disp is not None`, then it overrides the supplied version of `iprint` with the behaviour you outlined.

**Review by andyfaff — CHANGES_REQUESTED** [review]
@sciunto thanks for noticing this and submitting a PR. The comments I made were purely based on what the code does at the moment. I don't know why the function accepts both `iprint` and `disp` when a single parameter would do (possibly back compatibility).

**Comment by sciunto** [comment]
@andyfaff Thanks for helping. It's fixed and squashed.

**Comment by rgommers** [comment]
LGTM now, merged. Thanks @sciunto, @mdhaber

**Pull request merged**